src/server.c

```diff
@@ -1936,6 +1942,16 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
     /* Handle writes with pending output buffers. */
     handleClientsWithPendingWrites();
 
+    /* IO thread replica clients that are waiting for new replication data will
+     * stay indefinitely in main thread if there is no new data. We must make
+     * sure to send them to IO thread so as to give them chance to
+     * read the ACK message. Otherwise, they will be disconnected after
+     * repl_timeout. The below function only checks if IO thread has raised the
+     * flag that there is pending ACK message. Replica clients that send
+     * REPLCONF GETACK are immediately put in the pendingClientsToIOThreads
+     * queue inside sendGetackToReplicas */
+    sendReplicasToIOThread(1);
```

Comment:
I was thinking we will check if the replica client has replication buffer or there is a pending read event here, instead of separate places.